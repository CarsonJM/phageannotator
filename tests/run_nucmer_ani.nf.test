nextflow_pipeline {

    name "Test pipeline: NFCORE_PHAGEANNOTATOR"
    script "main.nf"
    tag "PIPELINE"


    test("run_nucmer_ani = true") {
        when {
            params {
                input           = "$projectDir/assets/test-datasets/samplesheets/two_samples_no_reads.csv"
                outdir          = "$outputDir"
                run_nucmer_ani  = true
                max_cpus        = 4
                max_memory      = '15.GB'
                max_time        = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${params.outdir}/ANICalculation/nucmer/")
                    ).match() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() }
            )
        }
    }
}
