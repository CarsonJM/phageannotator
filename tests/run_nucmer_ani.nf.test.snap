{
    "run_nucmer_ani = true": {
        "content": [
            [
                "all_samples.coords.tsv:md5,5b664ec6f6e7a9f609fc577f162464cc"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T16:12:34.856492165"
    }
}